secboot/secboot.go

```diff
@@ -29,11 +29,26 @@ import (
 	"github.com/snapcore/snapd/bootloader"
 )
 
+type LoadChain struct {
+	*bootloader.BootFile
+	Next []*LoadChain
+}
+
+// NewLoadChain returns a LoadChain corresponding to loading the given
+// BootFile before any of the given next chains.
+func NewLoadChain(bf bootloader.BootFile, next ...*LoadChain) *LoadChain {
```

Comment:
We don't have a scenario when there's more than one `next`, do we?

Reply:
you mean in this PR or in in practice? the answer yes to both

Reply:
the plan is instead of explode multiple boot asserts variants to boot chains, to generate directly trees from boot chains where Hashes can be more than one element sometimes

Reply:
I was confused by the tests a bit and had to checkout the branch locally. I've suggested a little tweak to formatting there that makes it more clear to see the chains and what is passed as alternative paths in `next`.